ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -642,7 +635,15 @@ let force_state_reset_keep_current_operations ~__context ~self ~value:state =
     Db.VM.set_resident_on ~__context ~self ~value:Ref.null;
     (* make sure we aren't reserving any memory for this VM *)
     Db.VM.set_scheduled_to_be_resident_on ~__context ~self ~value:Ref.null;
-    Db.VM.set_domid ~__context ~self ~value:(-1L)
+    Db.VM.set_domid ~__context ~self ~value:(-1L);
+
+    (* release vGPUs associated with VM *)
+    Db.VM.get_VGPUs ~__context ~self
+    |> List.iter (fun vgpu ->
+        Db.VGPU.set_currently_attached ~__context ~self:vgpu ~value:false;
```

Comment:
I think that this must stay in the Halted-only case, just like the `currently_attached` fields on the other device types. This means that `currently_attached` remains `true` on suspended VMs. It is just the resource reservation (the lines below) that need to be cleared on suspend.